internal/context.go

```diff
@@ -284,3 +322,89 @@ func createContextFileIfNotExist() {
 		log.Verboseln("Created ~/.corectl/contexts.yml for storage of corectl contexts")
 	}
 }
+
+func getSessionCookie(tlsClientConfig *tls.Config, engineURL string, userName string, password string) string {
+	// Verify Qlik Sense URL
+	u, err := url.Parse(engineURL)
+
+	if err != nil {
+		log.Fatalln("The engineURL doesn't seem to be correct")
+	}
+
+	if u.Scheme != "https" {
+		log.Fatalln("Only login through secure connections (HTTPS) is supported")
+	}
+
+	// Get username
+	if len(userName) == 0 {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Enter Username (domain\\user): ")
+		userName, _ = reader.ReadString('\n')
+	}
+
+	if !strings.Contains(userName, "\\") {
+		log.Fatalln("username MUST be in format 'domain\\user'")
+	}
+
+	// Get password
+	if len(password) == 0 {
+		fmt.Print("Enter Password: ")
+		bytePassword, _ := terminal.ReadPassword(int(syscall.Stdin))
+		password = string(bytePassword)
+	}
+
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsClientConfig
+
+	// Get post URL via redirects
+	resp, err := http.Get(engineURL)
+
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Generate xrfkey
+	xrfkey := generateXrfkey()
+
+	//
+	loginURL := resp.Request.URL
+	q, _ := url.ParseQuery(loginURL.RawQuery)
```

Comment:
nit: I think you can just do a `url.Query()` here.

Reply:
`Query not declared by package url`

Reply:
My bad, it was a sloppy comment!
I meant this one: https://golang.org/pkg/net/url/#URL.Query
```suggestion
	q := loginURL.Query()
```